crates/relayer/src/chain/cosmos/eip_base_fee.rs

```diff
@@ -5,17 +5,33 @@ use std::str::FromStr;
 use serde::Deserialize;
 use subtle_encoding::base64;
 use tendermint_rpc::Url;
-use tracing::debug;
+use tracing::{debug, trace};
 
-use ibc_proto::cosmos::base::v1beta1::DecProto;
+use ibc_proto::cosmos::base::v1beta1::{DecCoin, DecProto};
+use ibc_relayer_types::core::ics24_host::identifier::ChainId;
 
 use crate::error::Error;
 
-pub async fn query_eip_base_fee(rpc_address: &Url) -> Result<f64, Error> {
+pub async fn query_eip_base_fee(
+    rpc_address: &Url,
+    gas_price_denom: &str,
+    chain_id: &ChainId,
+) -> Result<f64, Error> {
     debug!("Querying Omosis EIP-1559 base fee from {rpc_address}");
 
-    let url =
-        format!("{rpc_address}/abci_query?path=\"/osmosis.txfees.v1beta1.Query/GetEipBaseFee\"");
+    let chain_name = chain_id.name();
+
+    let url = if chain_name.starts_with("osmosis") || chain_name.starts_with("osmo-test") {
```

Comment:
One last suggestion I missed earlier:

```suggestion
    let is_osmosis = chain_name.starts_with("osmosis") || chain_name.starts_with("osmo-test");

    let url = if is_osmosis {
```